### ISSUE TITLE
Allow configuring message length limit

### ISSUE THREAD

**Issue by berzi** [magmax/python-inquirer#312]
The documentation only says "if the message is too long it will be cut to fit". How long is too long? Why is this limit there? How can it be configured or removed?

It's rather frustrating when a library restricts usage with no apparent reason nor way to go around it. I need to show a hit of a couple lines to the user to help drive their decision on a question. If I had encountered this issue earlier in development I would have picked a different library.

**Comment by Cube707** [comment]
The rendere currenly can't handle the message breaking onto a second line. This would lead to only parts of the rendered question-type to be erased on re-render. Thats why the message wil be croped to fit. See:

https://github.com/magmax/python-inquirer/blob/b8f26f4968d0bca3d27e5322bc51ce4bc077b479/src/inquirer/render/console/__init__.py#L74

The width is determend by quering your terminal width. If it doesn't provide the information, the default of 80 is used, which is a long standing default width for terminals used since the beginning of linux. See:

https://github.com/magmax/python-inquirer/blob/b8f26f4968d0bca3d27e5322bc51ce4bc077b479/src/inquirer/render/console/__init__.py#L170-L172


We would be happy to accept a PR to enable the renderer to use multi line messages though...

**Comment by Cube707** [comment]
Duplicate of #99

See #115 #162 on what happens if we don't crop correctly

**Comment by Cube707** [comment]
See #115 #162 on what happens if we don't crop correctly

**Comment by berzi** [comment]
How come if i insert a \n in the message it gets rendered correctly then?
Can the renderer not force a \n at the 80 characters limit, as a workaround?

In any case i think the documentation should at least mention what the
limit is.

**Comment by Cube707** [comment]
What works for you? Using `Text`, `List` or `Checkbox` will leave the first line on screen with every rerender (so on every keyboard input). Here is an example:

```python
from pprint import pprint
import inquirer

questions = [
    inquirer.List(
        "size",
        message="What size do you need?\nSelct one",
        choices=["Jumbo", "Large", "Standard", "Medium", "Small", "Micro"],
    ),
]

answers = inquirer.prompt(questions)
pprint(answers)
```

fater pressung <kbd>UP</kbd>/<kbd>DOWN</kbd> a couple times, the result is:

![image](https://user-images.githubusercontent.com/62838959/208309558-47f32720-1218-4c5e-b740-676fa4baa64c.png)


As you can see, the first line appears multiple times

**Comment by Cube707** [comment]
To implement this featutre the render has to be reworked to allow it to split the message if its to long (withouot splitting in the middle of a word), remeneber how many lines it spans (while also allowing the programmer to manually use `\n`) and correctly indent every new line.

This has to be made compatible with existing code. BUT it would make sense to re-use this feature throught the rendere (for example for very long prompts, etc.). This requires even more dynamic... As you can see this is a lot of work and requires a bigger rewrighting of the rendere, so nobody has had time to do it.

Sidenote: There is also the resoning that super long Prompts usually are to much anyway and it is more userfriednly to split it up into multiple smaller questions or reformulate the question to be shorter, more on point.

**Comment by Cube707** [comment]
> In any case i think the documentation should at least mention what the
limit is.

How would this be documented other than it is currently? Anybody will just run the code to see if it fits anyway and not start counting characters. And as it depends on the specific termnal used, there is no point in providing a number.

**Comment by berzi** [comment]
> What works for you? Using `Text`, `List` or `Checkbox` will leave the first line on screen with every rerender (so on every keyboard input). Here is an example:
> 
> ```python
> from pprint import pprint
> import inquirer
> 
> questions = [
>     inquirer.List(
>         "size",
>         message="What size do you need?\nSelct one",
>         choices=["Jumbo", "Large", "Standard", "Medium", "Small", "Micro"],
>     ),
> ]
> 
> answers = inquirer.prompt(questions)
> pprint(answers)
> ```
> 
> after pressung UP/DOWN a couple times, the result is:
> 
> ![image](https://user-images.githubusercontent.com/62838959/208309558-47f32720-1218-4c5e-b740-676fa4baa64c.png)
> 
> As you can see, the first line appears multiple times

Oh, in fairness I had this issue with Confirm and only needed it for it, so i didn't notice the limitation.



> > In any case i think the documentation should at least mention what the
> > limit is.
> 
> How would this be documented other than it is currently? Anybody will just run the code to see if it fits anyway and not start counting characters. And as it depends on the specific terminal used, there is no point in providing a number.

I think mentioning "it will be trimmed to fit the width of the terminal" would be enough.



I'll take a look at the code more in depth over the holidays and consider a contribution (though don't count on it; if anyone wants to tackle it first don't let me stop you).

**Comment by Cube707** [comment]
> Oh, in fairness I had this issue with Confirm and only needed it for it, so i didn't notice the limitation.

Yes, if you only use yes/no you won't notice anything. But pressing any other key will also trigger the issue...

> ""

Decent Idea, will change the docs.
